users/views.py

```diff
@@ -211,32 +214,111 @@ def get_active_social_users(self, user):
             return []
         return UserSocialAuth.objects.filter(user=user)
 
+    def get_oidc_backends_end_session_url(self, request, social_users):
+        """Return end session url of the first OIDC backend the user has a
+        social auth entry on and hasn't been redirected yet.
+
+        The already redirected backends are tracked in a session variable because
+        in theory there could be multiple OIDC social auths for a user.
+        """
+        for social_user in social_users:
+            try:
+                backend_class = get_backend_class(social_user.provider)
+            except MissingBackend:
+                logger.warning(
+                    'Missing backend "{}" in user social auths for user {}'.format(
+                        social_user.provider,
+                        request.user.pk
+                    )
+                )
+                continue
+
+            if not issubclass(backend_class, OpenIdConnectAuth):
+                continue
+
+            backend = backend_class()
+
+            # Only redirect if it's enabled in the backend
+            if not backend.setting('REDIRECT_LOGOUT_TO_END_SESSION', default=False):
+                continue
+
+            end_session_endpoint = backend.oidc_config().get('end_session_endpoint')
+            if not end_session_endpoint:
+                continue
+
+            if 'oidc_backend_end_session_redirected' not in request.session:
+                request.session['oidc_backend_end_session_redirected'] = defaultdict(
+                    bool
+                )
+
+            if request.session['oidc_backend_end_session_redirected'].get(backend.name):
+                continue
```

Comment:
Not sure how the OIDC configs are cached, but should this check be more upwards, perhaps before the place where the `end_session_endpoint` is fetched?

Reply:
They are cached. Didn't want to set the variable if there is no end session endpoint.

Reply:
Makes sense.